### PR TITLE
changing google settings to run tests on device API 29

### PR DESCRIPTION
**Pull request by kkrishnareddy**
changed google settings so that we can run tests on API 29.

### CODE REVIEW

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 1]
```diff
@@ -210,4 +202,46 @@ public void activateAdmin() {
...
+            return UiAutomatorUtils.obtainChildInScrollable(
+                    "android:id/list",
+                    deviceAdmin.getAdminName()
+            );
+
+        return UiAutomatorUtils.obtainChildInScrollable(
+                "com.android.settings:id/recycler_view",
+                deviceAdmin.getAdminName()
+        );
+    }
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:243]: The code formatting doesn't match our current code style. Please follow the same code style as the one being used throughout the rest of the code base.

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 2]
```diff
@@ -210,4 +202,46 @@ public void activateAdmin() {
             Assert.fail(e.getMessage());
         }
     }
+
+    //obtaining date button in different devices.
+    UiObject changeDateButton()
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:207]: can be `private`

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 3]
```diff
@@ -210,4 +202,46 @@ public void activateAdmin() {
...
+    {
+        if(android.os.Build.VERSION.SDK_INT == 28)
+            return UiAutomatorUtils.obtainUiObjectWithText("Set date");
+
+        return UiAutomatorUtils.obtainEnabledUiObjectWithExactText("Date");
+    }
+
+    //obtaining button in a scrollable in different devices.
+    UiObject obtainButtonInScrollable(String Text)
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:216]: can be private

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 4]
```diff
@@ -210,4 +202,46 @@ public void activateAdmin() {
...
+            );
+
+        return UiAutomatorUtils.obtainChildInScrollable(
+                "com.android.settings:id/recycler_view",
+                Text
+        );
+    }
+
+    //obtaining disable admin button in different devices.
+    UiObject disableAdminButton(final DeviceAdmin deviceAdmin)
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:231]: can be private

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Left some comments

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java (hunk 1 above)
Comment (kkrishnareddy) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java]: reformated the code.

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java (hunk 2 above)
Comment (kkrishnareddy) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java]: changed to private.

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java (hunk 3 above)
Comment (kkrishnareddy) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java]: changed to private

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java (hunk 4 above)
Comment (kkrishnareddy) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java]: changed to private

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java (hunk 1 above)
Comment (shahzaibj) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java]: Please follow same code styles (braces etc..) as the one being used right now

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 5]
```diff
@@ -210,4 +202,51 @@ public void activateAdmin() {
             Assert.fail(e.getMessage());
         }
     }
+
+    /**
+     * @return returns the button of date in the settings screen.
+     */
+    private UiObject changeDateButton() {
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:209]:
```suggestion
    private UiObject obtainChangeDateButton() {
```

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 6]
```diff
@@ -210,4 +202,51 @@ public void activateAdmin() {
             Assert.fail(e.getMessage());
         }
     }
+
+    /**
+     * @return returns the button of date in the settings screen.
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:207]: Either provide complete javaodoc.....or just remove it as this is a private method

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 7]
```diff
@@ -210,4 +202,51 @@ public void activateAdmin() {
...
+     */
+    private UiObject changeDateButton() {
+        if (android.os.Build.VERSION.SDK_INT == 28)
+            return UiAutomatorUtils.obtainUiObjectWithText("Set date");
+
+        return UiAutomatorUtils.obtainEnabledUiObjectWithExactText("Date");
+    }
+
+    /**
+     * @param Text Text for which we need to find button in the UI.
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:217]: Either provide complete javaodoc.....or just remove it as this is a private method

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 8]
```diff
@@ -210,4 +202,51 @@ public void activateAdmin() {
...
+                    Text
+            );
+
+        return UiAutomatorUtils.obtainChildInScrollable(
+                "com.android.settings:id/recycler_view",
+                Text
+        );
+    }
+
+    /**
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:233]: Either provide complete javaodoc.....or just remove it as this is a private method

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java [hunk 9]
```diff
@@ -210,4 +202,51 @@ public void activateAdmin() {
...
+     * @param deviceAdmin deviceAdmin.getAdminName() for which we need to find button in the scrollable UI.
+     * @return the button of the device admin name in the UI.
+     */
+    private UiObject disableAdminButton(final DeviceAdmin deviceAdmin) {
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/device/settings/GoogleSettings.java:237]:
```suggestion
    private UiObject obtainDisableAdminButton(final DeviceAdmin deviceAdmin) {
```

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Left some comments

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**